libcaf_core/CMakeLists.txt

```diff
@@ -321,21 +223,139 @@ install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/caf"
 
 # -- build unit tests ----------------------------------------------------------
 
-if(NOT CAF_NO_UNIT_TESTS)
-  add_executable(caf-core-test
-                 test/core-test.cpp
-                 ${CAF_CORE_TEST_SOURCES}
-                 $<TARGET_OBJECTS:libcaf_core_obj>)
-  target_include_directories(caf-core-test PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/test")
-  target_compile_definitions(caf-core-test PRIVATE libcaf_core_EXPORTS)
-  target_link_libraries(caf-core-test ${CAF_EXTRA_LDFLAGS})
-  add_test_suites(caf-core-test
-                  "${CMAKE_CURRENT_SOURCE_DIR}"
-                  ${CAF_CORE_TEST_SOURCES})
+if(NOT CAF_ENABLE_TESTING)
+  return()
 endif()
 
-# -- add this library to the global CAF_LIBRARIES ------------------------------
+add_executable(caf-core-test
+  test/core-test.cpp
+  $<TARGET_OBJECTS:libcaf_core_obj>)
+
+target_include_directories(caf-core-test PRIVATE
+                           "${CMAKE_CURRENT_SOURCE_DIR}/test")
 
-list(APPEND CAF_LIBRARIES libcaf_core)
+target_compile_definitions(caf-core-test PRIVATE libcaf_core_EXPORTS)
 
-set(CAF_LIBRARIES ${CAF_LIBRARIES} PARENT_SCOPE)
+target_link_libraries(caf-core-test PUBLIC CAF::test)
+
+target_link_libraries(caf-core-test PRIVATE
+  $<TARGET_PROPERTY:libcaf_core,INTERFACE_LINK_LIBRARIES>)
+
+target_include_directories(caf-core-test PRIVATE
+                           "${CMAKE_BINARY_DIR}" "${CMAKE_CURRENT_SOURCE_DIR}")
```

Comment:
Why is `CMAKE_BINARY_DIR` included here?

Reply:
For finding `build_config.hpp`.